server/app/services/program/ProgramServiceImpl.java

```diff
@@ -1166,4 +1167,20 @@ private ProgramQuestionDefinition syncProgramQuestionDefinition(
     QuestionDefinition questionDefinition = roQuestionService.getQuestionDefinition(pqd.id());
     return pqd.loadCompletely(programDefinitionId, questionDefinition);
   }
+
+  /*
+   * Looks at the most recent version of each program and returns the program marked as the
+   * common intake form if it exists. The most recent version may be in the draft or active stage.
+   */
+  private Optional<ProgramDefinition> getCommonIntakeForm() {
+    ActiveAndDraftPrograms activeAndDraftPrograms = getActiveAndDraftPrograms();
+    for (String name : activeAndDraftPrograms.getProgramNames()) {
```

Comment:
idk if this suggestion makes sense, but:
What if, instead of getting a list of active and draft program names from `ActiveAndDraftPrograms` and then asking `ActiveAndDraftPrograms` if each of them is the most recent version, we added a `getMostRecentVersionsOfActiveAndDraftPrograms()` method to `ActiveAndDraftPrograms` that did that work for us?

Then this could look like
```
return getActiveAndDraftPrograms().getMostRecentVersionsOfActiveAndDraftPrograms().stream().findFirst(p -> p.isCommonIntakeForm());
```
since `findFirst()` returns an optional already.

Reply:
And I think that `getMostRecentProgramDefintions()` function could look like 
```java
  public ImmutableList<ProgramDefinition> getMostRecentProgramDefinitions() {
    return getProgramNames()
      .stream()
      .map(name -> getDraftProgramDefinition(name).orElseGet(getActiveProgramDefinition(name)::get))
      .collect(ImmutableList.toImmutableList());
  }
```

Reply:
Honestly I think they're pretty equivalent. Not doing that is slightly more efficient because it exits early without getting the definition for every name, but they have the same big O time so it's negligible overall. Changed it since you have a preference readability-wise.

Reply:
Yeah that's fair. It wasn't a strongly held preference, fwiw, but thanks for changing it 🙂